apps/src/bin/quiche-server.rs

```diff
@@ -187,129 +217,114 @@ fn main() {
             false => clients.values().filter_map(|c| c.conn.timeout()).min(),
         };
 
+        trace!("Polling the event loop for I/O events");
         poll.poll(&mut events, timeout).unwrap();
 
-        // Read incoming UDP packets from the socket and feed them to quiche,
-        // until there are no more packets to read.
-        'read: loop {
-            // If the event loop reported no events, it means that the timeout
-            // has expired, so handle it without attempting to read packets. We
-            // will then proceed with the send loop.
-            if events.is_empty() && !continue_write {
-                trace!("timed out");
-
-                clients.values_mut().for_each(|c| c.conn.on_timeout());
-
-                break 'read;
-            }
+        // If the event loop reported no events, it means that the timeout
+        // has expired, so handle it without attempting to read packets. We
+        // will then proceed with the send loop.
+        if events.is_empty() && !continue_write {
+            trace!("timed out");
 
-            let (len, from) = match socket.recv_from(&mut buf) {
-                Ok(v) => v,
-
-                Err(e) => {
-                    // There are no more UDP packets to read, so end the read
-                    // loop.
-                    if e.kind() == std::io::ErrorKind::WouldBlock {
-                        trace!("recv() would block");
-                        break 'read;
-                    }
+            clients.values_mut().for_each(|c| c.conn.on_timeout());
+        }
 
-                    panic!("recv() failed: {:?}", e);
+        // Part 1/3:
+        // Read incoming UDP packets from the socket and feed them to quiche,
+        // until there are no more packets to read.
+        for event in &events {
+            // Match the event to the correct socket
+            let socket = match event.token() {
+                mio::Token(0) => {
+                    trace!("Event for starting socket occurred");
+                    &starting_socket
                 },
-            };
 
-            trace!("got {} bytes", len);
+                mio::Token(1) => {
+                    trace!("Event for preferred socket occured: {:?}", event);
+                    &preferred_socket
+                },
 
-            let pkt_buf = &mut buf[..len];
+                _ => unreachable!(),
+            };
 
-            if let Some(target_path) = conn_args.dump_packet_path.as_ref() {
-                let path = format!("{target_path}/{pkt_count}.pkt");
+            let local_addr = socket.local_addr().unwrap();
 
-                if let Ok(f) = std::fs::File::create(path) {
-                    let mut f = std::io::BufWriter::new(f);
-                    f.write_all(pkt_buf).ok();
-                }
-            }
+            trace!("Read loop starting up for {:?}", socket);
```

Comment:
are we able to qualify which socket is being used here, or elsewhere? I think that will be useful to avoid any confusion

Reply:
Added the socket name to the trace 👍